android/src/main/java/com/theoplayer/media/MediaPlaybackService.kt

```diff
@@ -208,53 +199,16 @@ class MediaPlaybackService : MediaBrowserServiceCompat() {
     mediaSessionConnector.removeListener(mediaSessionListener)
   }
 
-  override fun onGetRoot(
-    clientPackageName: String,
-    clientUid: Int,
-    rootHints: Bundle?
-  ): BrowserRoot {
-    // (Optional) Control the level of access for the specified package name.
-    // You'll need to write your own logic to do this.
-    return if (allowBrowsing(clientPackageName, clientUid)) {
-      // Returns a root ID that clients can use with onLoadChildren() to retrieve
-      // the content hierarchy.
-      BrowserRoot(BROWSABLE_ROOT, null)
-    } else {
-      // Clients can connect, but this BrowserRoot is an empty hierachy
-      // so onLoadChildren returns nothing. This disables the ability to browse for content.
-      BrowserRoot(EMPTY_ROOT, null)
-    }
-  }
-
-  @Suppress("UNUSED_PARAMETER")
-  private fun allowBrowsing(clientPackageName: String, clientUid: Int): Boolean {
-    // Only allow browsing from the same package
-    return TextUtils.equals(clientPackageName, packageName)
-  }
-
-  override fun onLoadChildren(
-    parentId: String,
-    result: Result<List<MediaBrowserCompat.MediaItem>>
-  ) {
-    if (parentId == EMPTY_ROOT) {
-      result.sendResult(null)
-      return
-    }
-    result.sendResult(emptyList())
-  }
-
   private fun updateNotification() {
-    player?.let {
-      if (it.isPaused) {
-        updateNotification(PlaybackStateCompat.STATE_PAUSED)
-      } else {
-        updateNotification(PlaybackStateCompat.STATE_PLAYING)
-      }
+    val player = player
+    when {
+      player?.source == null -> updateNotification(PlaybackStateCompat.STATE_STOPPED)
+      !player.isPaused -> updateNotification(PlaybackStateCompat.STATE_PLAYING)
```

Comment:
```suggestion
      it.source == null -> updateNotification(PlaybackStateCompat.STATE_STOPPED)
      it.isPaused -> updateNotification(PlaybackStateCompat.STATE_PLAYING)
```

Reply:
@tvanlaerhoven  sorry, it makes sense, consider it as a nitpick. marking as resolved and leaving it up to you.

Reply:
it might be good to also stop the notification here in case an update of the notification is requested, but somehow the player is null (not sure if that could even happen), which is covered by this case:

`player?.source == null -> updateNotification(PlaybackStateCompat.STATE_STOPPED)`

Reply:
it would be useful to have this protection in the SDK from my point.